### PR TITLE
Add immutable updates on access and authorize tokens

### PR DESCRIPTION
**Pull request by smarterclayton**
Still need to add tests to verify this is actually not mutating since we have none.

[test]

@liggitt to support migrate

### CODE REVIEW

File: pkg/oauth/registry/oauthaccesstoken/strategy.go [hunk 1]
```diff
@@ -57,6 +62,11 @@ func (s strategy) Validate(ctx kapi.Context, obj runtime.Object) field.ErrorList
 	return validationErrors
 }
 
+// ValidateUpdate validates an update
+func (s strategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
+	return kvalidation.ValidateImmutableField(obj, old, &field.Path{})
```
Comment (liggitt) [inline pkg/oauth/registry/oauthaccesstoken/strategy.go]: Actually want a validate update method that checks the fields we want immutable. Doing it this way means things like uid, creationTimestamp, etc, aren't defaulted from the existing object, which is inconsistent with all other objects

File: pkg/oauth/registry/oauthaccesstoken/etcd/etcd.go [hunk 2]
```diff
@@ -99,6 +100,10 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 	return r.store.Create(ctx, obj)
 }
 
+func (r *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	return r.store.Update(ctx, obj)
```
Comment (liggitt) [inline pkg/oauth/registry/oauthaccesstoken/etcd/etcd.go]: No real reason to explicitly delegate now that we're allowing update

**Comment by liggitt** [comment]
Adding update support requires regenerating swagger spec for these types

File: pkg/oauth/registry/oauthaccesstoken/strategy.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/oauth/registry/oauthaccesstoken/strategy.go]: I though all that defaulting was removed?

Comment (liggitt) [inline pkg/oauth/registry/oauthaccesstoken/strategy.go]: No, it lives on in ValidateObjectMetaUpdate for compatibility with v1

Comment (smarterclayton) [inline pkg/oauth/registry/oauthaccesstoken/strategy.go]: K.  Although this is net new and it's one less thing we have to worry
about, I don't care either way.

**Comment by smarterclayton** [comment]
[test] yum

**Comment by smarterclayton** [comment]
Now with tests, ready for final review

**Comment by liggitt** [comment]
needs hack/update-generated-swagger-spec.sh, LGTM

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**